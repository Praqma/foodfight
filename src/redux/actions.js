import { ADD_FOOD } from "./actionTypes";

export const addFood = content => ({
    type: ADD_FOOD,
    payload: {
        content
    }
  });