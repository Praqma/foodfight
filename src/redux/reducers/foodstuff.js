import { ADD_FOOD } from "../actionTypes";

const initialState = {
   allFood: []
};

export default function(state = initialState, action) {
  switch (action.type) {
    case ADD_FOOD: {
      const { name } = action.payload;
      return {
        ...state,
          allFood: [...state.allFood, name]        
      };  
    }
    default:
      return state;
  }
}
