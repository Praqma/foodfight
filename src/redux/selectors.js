export const getAllFood = store => store.foodstuff;