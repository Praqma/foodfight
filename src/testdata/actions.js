import { ADD_FOOD } from '../redux/actionTypes';

export const addPotato = {
    type: ADD_FOOD,
    payload: {name:'potato'}
  };