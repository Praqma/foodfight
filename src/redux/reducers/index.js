
import { combineReducers } from "redux";
import foodstuff from "./foodstuff";

export default combineReducers({ foodstuff });