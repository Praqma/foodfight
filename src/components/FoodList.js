import React from "react";
import { connect } from "react-redux";
import Food from "./Food";
import {getAllFood} from "../redux/selectors"
const FoodList = ({allFood}) => (
    
        <ul>
            {allFood && allFood.length
            ? allFood.map((food, index) => {
                return <li key={food}><Food name={food}/></li>
            })
        : "You need a more varied diet!"}
        </ul>
);

export function mapStateToProps(state) {
    return getAllFood(state)
  
};
export default  connect(mapStateToProps)(FoodList);
