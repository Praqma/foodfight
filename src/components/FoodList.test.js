import React from 'react';
import FoodList, {mapStateToProps} from './FoodList';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);
import { addFood } from '../redux/actions'
import foodstuff from '../redux/reducers/foodstuff';

test('renders the list', () => {
    let store = mockStore({foodstuff:{allFood: []}});    

    const component  = renderer.create(
        <Provider store={store}><FoodList/></Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})  

test('renders the list with items', () => {
    let props = {allFood: ['apple','potato']}
    let store = mockStore({foodstuff: props});    
    const component  = renderer.create(
        <Provider store={store}><FoodList/></Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})  

test('mapStateToProps', () => {
    let state = {foodstuff:{allFood: ['apple','potato']}};
    expect(mapStateToProps(state)).toEqual({allFood: ['apple','potato']});
}); 
