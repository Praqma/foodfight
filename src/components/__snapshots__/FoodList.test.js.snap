// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the list 1`] = `
<ul>
  You need a more varied diet!
</ul>
`;

exports[`renders the list with items 1`] = `
<ul>
  <li>
    <p>
      apple
    </p>
  </li>
  <li>
    <p>
      potato
    </p>
  </li>
</ul>
`;
